internal/extender/resourcereservations.go

```diff
@@ -15,26 +15,281 @@
+// or a soft reservation if dynamic allocation is enabled.
+func (rrm *ResourceReservationManager) ReserveForExecutor(ctx context.Context, executor *v1.Pod, node string) error {
+	rrm.mutex.Lock()
+	defer rrm.mutex.Unlock()
+
+	// TODO: make sure executor doesn't already have a reservation, if it does, it must be freed (we don't seem to handle this in the current logic, but we should going forward)
+
+	unboundReservationsToNodes, err := rrm.getUnboundReservations(ctx, executor)
+	if err != nil {
+		return err
+	}
+	// Sort the unbound reservations such that we favor reservations that are already tied to the requested node first
+	sortedUnboundReservations := make([]string, len(unboundReservationsToNodes))
+	for reservation := range unboundReservationsToNodes {
+		sortedUnboundReservations = append(sortedUnboundReservations, reservation)
+	}
+	sort.Slice(sortedUnboundReservations, func(i, j int) bool {
+		return unboundReservationsToNodes[sortedUnboundReservations[i]] == node
+	})
+
+	if len(sortedUnboundReservations) > 0 {
+		reservationName := sortedUnboundReservations[0]
+		return rrm.bindExecutorToResourceReservation(ctx, executor, reservationName, unboundReservationsToNodes[reservationName])
+	}
+
+	// Try to get a soft reservation if it is a dynamic allocation application
+	extraExecutorFreeSpots, err := rrm.getFreeExtraExecutorSpots(ctx, executor)
+	if err != nil {
+		return werror.WrapWithContextParams(ctx, err, "failed to count free extra executor spots remaining")
+	}
+	if extraExecutorFreeSpots > 0 {
+		return rrm.bindExecutorToSoftReservation(ctx, executor, node)
+	}
+
+	return werror.ErrorWithContextParams(ctx, "failed to find free reservation for executor")
+}
+
+// GetReservedResources returns the resources per node that are reserved for executors.
+func (rrm *ResourceReservationManager) GetReservedResources() resources.NodeGroupResources {
+	resourceReservations := rrm.resourceReservations.List()
+	usage := resources.UsageForNodes(resourceReservations)
+	usage.Add(rrm.softReservationStore.UsedSoftReservationResources())
+	return usage
+}
+
+func (rrm *ResourceReservationManager) bindExecutorToResourceReservation(ctx context.Context, executor *v1.Pod, reservationName string, node string) error {
+	resourceReservation, ok := rrm.GetResourceReservation(executor)
+	if !ok {
+		return werror.ErrorWithContextParams(ctx, "failed to get resource reservationName")
+	}
+	copyResourceReservation := resourceReservation.DeepCopy()
+	reservationObject := copyResourceReservation.Spec.Reservations[reservationName]
+	reservationObject.Node = node
+	copyResourceReservation.Spec.Reservations[reservationName] = reservationObject
+	copyResourceReservation.Status.Pods[reservationName] = executor.Name
+	err := rrm.resourceReservations.Update(copyResourceReservation)
+	if err != nil {
+		return werror.WrapWithContextParams(ctx, err, "failed to update resource reservationName")
```

Comment:
pass reservationName as a param

Reply:
actually we can probably not wrap and directly return the result of the `Update` call, as we also wrap the error at `resource.go`

Reply:
The extra message might be clearer when wrapping.